examples/sensors/imu.py

```
@@ -0,0 +1,139 @@
+    elif IS_MATPLOTLIB_AVAILABLE:
+        imu.add_recording(
+            gs.options.RecordingOptions(
+                handler=MPLPlotter(title="IMU Measured Data", labels=labels),
+                run_in_thread=False,
+            )
+        )
+        imu.add_recording(
+            gs.options.RecordingOptions(
+                handler=MPLPlotter(title="IMU Ground Truth Data", labels=labels),
+                run_in_thread=False,
+            ),
+            read_ground_truth=True,
+        )
+    else:
+        print("matplotlib or pyqtgraph not found, skipping real-time plotting.")
+
+    imu.add_recording(
+        gs.options.RecordingOptions(
+            handler=NPZFileWriter(filename="imu_data.npz"),
+        )
+    )
+    ########################## build ##########################
+    scene.build()
+
+    franka.set_dofs_kp(
+        np.array([4500, 4500, 3500, 3500, 2000, 2000, 2000, 100, 100]),
+    )
+    franka.set_dofs_kv(
+        np.array([450, 450, 350, 350, 200, 200, 200, 10, 10]),
+    )
+    franka.set_dofs_force_range(
+        np.array([-87, -87, -87, -87, -12, -12, -12, -100, -100]),
+        np.array([87, 87, 87, 87, 12, 12, 12, 100, 100]),
+    )
+
+    circle_center = np.array([0.4, 0.0, 0.5])
+    circle_radius = 0.15
+    rate = 2 / 180 * np.pi
+
+    def control_franka_circle_path(i):
+        pos = circle_center + np.array([np.cos(i * rate), np.sin(i * rate), 0]) * circle_radius
+        qpos = franka.inverse_kinematics(
+            link=end_effector,
+            pos=pos,
+            quat=np.array([0, 1, 0, 0]),
+        )
+        franka.control_dofs_position(qpos[:-2], motors_dof)
+        scene.draw_debug_sphere(pos, radius=0.01, color=(1.0, 0.0, 0.0, 0.5))
+
+    try:
+        for i in tqdm(range(1000)):
+            scene.step()
+            control_franka_circle_path(i)
+    except KeyboardInterrupt:
+        gs.logger.info("Simulation interrupted, exiting.")
+    except gs.GenesisException as e:
+        gs.logger.error(e)
```

It would remove this branch. It is too likely to hide actual failure cases on user machine that would be harder to debug.